kvmagent/kvmagent/test/host_testsuite/test_host_facts.py

```
@@ -0,0 +1,57 @@
+                return "not implemented"
+
+        mock_call.side_effect = mock_shell_call
+
+        rsp_json_str = host.fact(None)
+        rsp = jsonobject.loads(rsp_json_str)
+
+        # test dmidecode's output
+        self.assertNotEqual(rsp.systemSerialNumber, 'unknown')
+
+        # change to error chars
+        error_char = open('kvmagent/test/host_testsuite/mock_error_chars', 'r').read()
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)